src/BadgeVendor.sol

```diff
@@ -2,32 +2,27 @@
 pragma solidity ^0.8.6;
 
 import {Badges} from "./Badges.sol";
-import {ChainClaim} from "chain-claim/ChainClaim.sol";
 
-contract BadgeVendor is ChainClaim {
-  constructor(
-    address issuer,
-    string memory name
-  ) ChainClaim(issuer, name) {}
+contract BadgeVendor {
+  Badges _badges;
 
-	function _genDataHash(
-		address chainedAddress
-	) external view returns (bytes32) {
-		return genDataHash(chainedAddress);
+  constructor (
+    address badgesAddress
+  ) {
+    _badges = Badges(badgesAddress);
   }
 
-  function takeBadge(
-    address collection,
-    address issuedAddress,
-    uint8[2] memory v,
-    bytes32[2] memory r,
-    bytes32[2] memory s
+  function claimBadge(
+    address fromAddress,
+    string calldata tokenURI,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
   ) external returns (uint256) {
-    require(
-      claim(issuedAddress, msg.sender, v, r, s),
-      "BadgeVendor: Failed taking Badge"
-    );
-    Badges badges = Badges(collection);
-    return badges.mint(msg.sender);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    uint256 tokenId = _badges.mintWithPermission(fromAddress, tokenURI, signature);
```

Comment:
The problem I see here, is that really `mintWithPermission` should be called from the user.

Otherwise, since `mintWithPermission` uses "msg.sender=to`, we're creating a relationship between

from=fromAddress (which will be e.g. an Otterspace representative)
to=BadgeVendor

whereas to=user's address